### PR TITLE
Feature/add api logout

### PR DESCRIPTION
**Pull request by TanNguyen17112003**

### CODE REVIEW

**Review by ngyngcphu — CHANGES_REQUESTED** [review]
Please resolve conflicts.

File: src/components/common/DesktopNavbar.tsx [hunk 1]
```diff
@@ -184,6 +184,47 @@ export const DesktopNavbar: Component<{ menu: RouteMenu }> = ({ menu }) => {
...
+                          Tania Andrew
+                        </Typography>
+                        <Typography variant='small' color='gray' className='font-normal'>
+                          {data ? data.email : ''}
+                        </Typography>
+                      </div>
+                    </div>
+                    <hr className='my-1' />
+                    <List className='p-0'>
+                      <ListItem onClick={() => emitEvent('logout')}>{menuItem.name}</ListItem>
```
Comment (ngyngcphu) [inline src/components/common/DesktopNavbar.tsx]: Duplicate code. `emitEvent` was used in `App.tsx`. Use attribute `onClick` of `menuItem` instead.

File: src/components/common/SidebarMenu.tsx [hunk 2]
```diff
@@ -165,7 +165,11 @@ export function useSidebarMenu() {
                   }
                   if (isSuccess && menuItem.type === 'logout-btn') {
                     return (
-                      <ListItem key={idx} className={SIDEBAR_ITEM_CLASSNAME}>
+                      <ListItem
+                        key={idx}
+                        className={SIDEBAR_ITEM_CLASSNAME}
+                        onClick={() => emitEvent('logout')}
```
Comment (ngyngcphu) [inline src/components/common/SidebarMenu.tsx]: Also duplicate code.

File: src/layouts/AppLayout.tsx [hunk 3]
```diff
@@ -1,8 +1,22 @@
...
+import { useListenEvent } from '@hooks';
+import { useAuthMutation } from '@hooks';
 
 export const AppLayout: Component<{ menu: RouteMenu; child: RouteChild }> = ({ menu, child }) => {
+  const [openDialog, setOpenDialog] = useState<boolean>(false);
+  const handleOpenDialog = () => {
+    setOpenDialog(!openDialog);
+  };
+  const { logout } = useAuthMutation();
+  const handleLogout = () => {
```
Comment (ngyngcphu) [inline src/layouts/AppLayout.tsx]: WTF? `logout.mutateAsync` returns a Promise. You must use `async-await`.

**Review by ngyngcphu — CHANGES_REQUESTED** [review]

File: src/components/common/DesktopNavbar.tsx (hunk 1 above)
Comment (TanNguyen17112003) [reply src/components/common/DesktopNavbar.tsx]: image
I test with <ListItem onClick={menuItem.onClick}>{menuItem.name}</ListItem> but it didn't work!

Comment (ngyngcphu) [reply src/components/common/DesktopNavbar.tsx]: Why? I tested it and it worked.
image

File: src/layouts/AppLayout.tsx (hunk 3 above)
Comment (TanNguyen17112003) [reply src/layouts/AppLayout.tsx]: image

**Review by ngyngcphu — APPROVED** [review]
LGTM

**Pull request merged**